src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfoOfT.WriteHelpers.cs

```diff
@@ -76,8 +93,9 @@ public partial class JsonTypeInfo<T>
                 Debug.Assert(CanUseSerializeHandler);
                 Debug.Assert(Converter is JsonMetadataServicesConverter<T>);
 
-                using var bufferWriter = new PooledByteBufferWriter(Options.DefaultBufferSize);
-                Utf8JsonWriter writer = Utf8JsonWriterCache.RentWriter(Options, bufferWriter);
+                // Bufferwriter must be disposed after Utf8JsonWriter, use an unnamed variable to use cleaner using syntax.
```

Comment:
If order of disposals is significant, I would rather just make it explicit in a plain old `finally` clause.